### PR TITLE
Fix/pipeline apitest cost time

### PR DESCRIPTION
**Pull request by chengjoey**
#### What type of this PR

Add one of the following kinds:
/kind bug

#### What this PR does / why we need it:
use goroutine execute apitest make cost time current

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/all?id=237896&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sInN0YXRlcyI6WzQ0MDIsNzEwNCw3MTA1LDQ0MDMsNDQwNCw3MTA2LDQ0MDYsNDQwNyw0NDEyLDQ1MzgsNDQxMyw0NDE0LDQ0MTUsNDQxNl0sImFzc2lnbmVlSURzIjpbIjEwMDEyMDUiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=541&type=BUG)

#### Specified Reviewers:

/assign @your-reviewer

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/pipengine/reconciler/taskrun/taskrun.go [hunk 1]
```diff
@@ -70,8 +71,10 @@ func New(ctx context.Context, task *spec.PipelineTask,
 	actionAgentSvc *actionagentsvc.ActionAgentSvc,
 	extMarketSvc *extmarketsvc.ExtMarketSvc,
 ) *TaskRun {
+	// make executor has buffer, don't block task framework
+	executorCH := make(chan interface{}, 1)
```
Comment (sfwn) [inline modules/pipeline/pipengine/reconciler/taskrun/taskrun.go:75]: executorCh

File: modules/pipeline/pipengine/actionexecutor/plugins/apitest/apitest.go [hunk 2]
```diff
@@ -28,9 +31,10 @@ import (
 var Kind = types.Kind(spec.PipelineTaskExecutorKindAPITest)
 
 type define struct {
-	name     types.Name
-	options  map[string]string
-	dbClient *dbclient.Client
+	name        types.Name
+	options     map[string]string
+	dbClient    *dbclient.Client
+	runningApis sync.Map
```
Comment (sfwn) [inline modules/pipeline/pipengine/actionexecutor/plugins/apitest/apitest.go:37]: runningAPIs

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
add screenshot for your tests.

**Comment by chengjoey** [comment]
> add screenshot for your tests.

image

**Review by sfwn — APPROVED** [review]

**Pull request merged**